tools/mlir-gen/MLIRGen.cpp

```diff
@@ -340,7 +338,7 @@ Value MLIRGenerator::lowerRelu(Value input, Value output) {
   if (!enableRelu)
     return input;
 
-  auto zero = getConstFloat(builder, 0.0, dataType.getIntOrFloatBitWidth());
+  auto zero = getConstFloat(builder, 0.0, dataType.cast<FloatType>());
```

Comment:
Isn't this always going to create the constant at `f32`? Won't that cause casts to appear?

Reply:
No, `FloatType` carries the info about the specific float type f8, f16, f32, bf16 etc.
That is correctly propagated and the created constant is of the specified `dataType`